File: fix_env.js
const fs = require('fs');
const path = require('path');

const envPath = path.join(__dirname, '.env.local');

try {
    let content = fs.readFileSync(envPath);

    // Convert buffer to string, filtering out null bytes usually associated with UTF-16LE interpreted as UTF-8
    // or just mixed encoding artifacts.
    let stringContent = content.toString('utf8').replace(/\0/g, '');

    // Normalize newlines
    stringContent = stringContent.replace(/\r\n/g, '\n').replace(/\r/g, '\n');

    fs.writeFileSync(envPath, stringContent, { encoding: 'utf8' });
    console.log('✅ .env.local encoding fixed/cleaned successfully.');
} catch (error) {
    console.error('❌ Error fixing .env.local:', error);
}

File: fix_env.py
import os

file_path = '.env.local'

try:
    if os.path.exists(file_path):
        # Read as binary to capture all bytes
        with open(file_path, 'rb') as f:
            content = f.read()
        
        # Remove null bytes
        clean_content = content.replace(b'\x00', b'')
        
        # Decode trying utf-8, fallback to latin-1 if needed, correcting CRLF
        text = clean_content.decode('utf-8', errors='ignore').replace('\r\n', '\n')
        
        # Write back as pure UTF-8
        with open(file_path, 'w', encoding='utf-8') as f:
            f.write(text)
            
        print("✅ .env.local cleaned successfully (Python).")
    else:
        print("⚠️ File not found.")

except Exception as e:
    print(f"❌ Error: {e}")
